homeassistant/components/binary_sensor/homematicip_cloud.py

```diff
@@ -95,7 +95,7 @@ def device_class(self):
     @property
     def is_on(self):
         """Return true if smoke is detected."""
-        return self._device.smokeDetectorAlarmType != STATE_SMOKE_OFF
+        return str(self._device.smokeDetectorAlarmType) != STATE_SMOKE_OFF
```

Comment:
If it's an enum, shouldn't we use the name or value attribute when comparing?

Reply:
I've changed it to an enum comparison now.